dipy/align/scalespace.py

```diff
@@ -0,0 +1,422 @@
+        requested level
+
+        """
+        return self._get_attribute(self.scalings, level)
+
+    def get_affine(self, level):
+        r"""Voxel-to-space transformation at a given level
+
+        Returns the voxel-to-space transformation associated to the sub-sampled
+        image at a particular resolution of the scale space (note that this
+        object does not explicitly subsample the smoothed images, but only
+        provides the properties the sub-sampled images must have).
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get affine transform from
+
+        Returns
+        -------
+            the affine (voxel-to-space) transform at the requested resolution
+            or None if an invalid level was requested
+        """
+        return self._get_attribute(self.affines, level)
+
+    def get_affine_inv(self, level):
+        r"""Space-to-voxel transformation at a given level
+
+        Returns the space-to-voxel transformation associated to the sub-sampled
+        image at a particular resolution of the scale space (note that this
+        object does not explicitly subsample the smoothed images, but only
+        provides the properties the sub-sampled images must have).
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get the inverse transform from
+
+        Returns
+        -------
+        the inverse (space-to-voxel) transform at the requested resolution or
+        None if an invalid level was requested
+        """
+        return self._get_attribute(self.affine_invs, level)
+
+    def get_sigmas(self, level):
+        r"""Smoothing parameters used at a given level
+
+        Returns the smoothing parameters (a scalar for each axis) used at the
+        requested level of the scale space
+
+        Parameters
+        ----------
+        level : int, 0 <= from_level < L, (L = number of resolutions)
+            the scale space level to get the smoothing parameters from
+
+        Returns
+        -------
+        the smoothing parameters at the requested level
+
+        """
+        return self._get_attribute(self.sigmas, level)
+
+
+class IsotropicScaleSpace(ScaleSpace):
+    def __init__(self, image, factors, sigmas,
+                 image_grid2world=None,
+                 input_spacing=None,
+                 mask0=False):
+        r""" IsotropicScaleSpace
+
+        Computes the Scale Space representation of an image using isotropic
+        smoothing kernels for all scales. The scale space is simply a list
+        of images produced by smoothing the input image with a Gaussian
+        kernel with different smoothing parameters.
+
+        This specialization of ScaleSpace allows the user to provide custom
+        scale and smoothing factors for all scales.
+
+        Parameters
+        ----------
```

Comment:
Again, which are optional and what are their defaults (especially where the kwarg default value is None, it's not obvious what that would entail!)